junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -141,4 +141,134 @@ private static boolean hasParameters(Method method) {
+			String indicativeName = classReplaceToIndicativeSentence(nestedClass);
+			return indicativeName;
+		}
+
+		@Override
+		public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
+			String methodName = underScoreNameFormatting(super.generateDisplayNameForMethod(testClass, testMethod));
+			String indicativeName = classReplaceToIndicativeSentence(testClass);
+			return indicativeName + ", " + methodName;
+		}
+
+		private String classReplaceToIndicativeSentence(Class<?> testClass) {
+			String classIndicativeSentence = null;
+			Class<?> classWithEnclosingParent = testClass.getEnclosingClass();
+			DisplayName classWithDisplayName = testClass.getAnnotation(DisplayName.class);
+			DisplayNameGeneration classWithAnnotation = testClass.getAnnotation(DisplayNameGeneration.class);
+
+			if (classWithEnclosingParent == null) {
+				if (classWithDisplayName != null)
+					classIndicativeSentence = classWithDisplayName.value();
+				else
+					classIndicativeSentence = underScoreNameFormatting(generateDisplayNameForClass(testClass));
+			}
+			else {
+				if (classWithAnnotation != null) {
+					if (classWithAnnotation.value() == IndicativeSentencesGenerator.class) {
+						if (classWithDisplayName != null)
+							classIndicativeSentence = classWithDisplayName.value();
+						else
+							classIndicativeSentence = underScoreNameFormatting(generateDisplayNameForClass(testClass));
+					}
+				}
+				else {
+					if (classWithDisplayName != null)
+						classIndicativeSentence = classWithDisplayName.value();
+					else
+						classIndicativeSentence = underScoreNameFormatting(
+							super.generateDisplayNameForNestedClass(testClass));
+
+					classIndicativeSentence = classReplaceToIndicativeSentence(classWithEnclosingParent) + ", "
+							+ classIndicativeSentence;
+				}
+			}
+
+			return classIndicativeSentence;
+		}
+
+		/**
+		 * Generate a string with simply the name of the test name without the
+		 * classes related divided by a $: AnotherParent$ParentName$Class.
+		 *
+		 * @param testName the Test Class from to extract the parameter types from;
+		 * never {@code blank}.
+		 * @return a string without the $ symbol and the root data that comes from
+		 * {@code Class.getName()}.
+		 */
+		private String removeRootCharacter(String testName) {
+			Preconditions.notBlank(testName, "Input Name parameter should not be blank");
+			int indexOfChar = testName.indexOf('$');
+
+			if (testName.indexOf('$') >= 0)
+				testName = testName.substring(indexOfChar + 1);
+
+			return testName;
+		}
+
+		/**
+		 * Generate a string which runs through the test name of a name parsed
+		 * by {@link ReplaceUnderscores} and removes the uppercase of the first letter.
+		 *
+		 * @param testName the Test Class from to extract the parameter types from;
+		 * never {@code blank}.
+		 * @return a string without underscore conversion capital letters.
+		 */
+		private String underScoreNameFormatting(String testName) {
```

Comment:
Do we really want to do that? I'm not sure we should go in such a level of detail...

Reply:
The purpose of these methods is for,
maintaining integrity when variations appear with the use of @DisplayName and if you write an underscore convention with uppercase letter at the beginning or in all except the first one, 
example,


```
@DisplayName("A stack")
class TestCase { ...

  @Nested
  class After_pushing_an_element_to_an_empty_stack {

    @Test
    void throws_EmptyStackException_When_Peeked() 
}

----> A stack
--------> A stack, After pushing an element to an empty stack
----------->A stack, After pushing an element to an empty stack, throws EmptyStackException When Peeked 
```
That it looks a kind of strange the output with uppercase letters the first letter in some words and in others not.

Anyway, if you still consider this formatting is too much, with out them the output will look like the example in that test and obviosuly the purpose of the feature would not be affected.

Reply:
I agree with @juliette-derancourt. Let's not do any capitalization changes.

Reply:
All right, the next commit will have the changes to revert that and also in the DisplayNamesTest file.